subxt/src/backend/mod.rs

```diff
@@ -336,54 +336,96 @@ pub struct StorageResponse {
-        type Item = RpcResult<String>;
+        pub type SubscriptionHandler = Box<
+            dyn for<'a> Fn(
+                    &'a mut MockDataTable,
+                    &'a mut Option<Subscription>,
+                    Option<Box<serde_json::value::RawValue>>,
+                )
+                    -> super::rpc::RawRpcFuture<'a, super::rpc::RawRpcSubscription>
+                + Send,
+        >;
+
+        pub type MethodHandler = Box<
+            dyn for<'a> Fn(
+                    &'a mut MockDataTable,
+                    &'a mut Option<Subscription>,
+                    Option<Box<serde_json::value::RawValue>>,
+                )
+                    -> super::rpc::RawRpcFuture<'a, Box<serde_json::value::RawValue>>
+                + Send,
+        >;
 
-        struct MockDataTable {
-            items: HashMap<Vec<u8>, VecDeque<Item>>,
+        pub enum Message<T> {
+            Many(RpcResult<Vec<T>>),
+            Single(T),
         }
 
-        impl MockDataTable {
-            fn new() -> Self {
-                MockDataTable {
-                    items: HashMap::new(),
+        impl<T> Message<T> {
+            pub fn single(self) -> T {
+                match self {
+                    Self::Single(s) => s,
+                    _ => todo!(),
+                }
+            }
+            pub fn many(self) -> RpcResult<Vec<T>> {
+                match self {
+                    Self::Many(s) => s,
+                    _ => todo!(),
                 }
             }
+        }
 
-            fn from_iter<'a, T: Serialize, I: IntoIterator<Item = (&'a str, RpcResult<T>)>>(
-                item: I,
-            ) -> Self {
-                let mut data = Self::new();
-                for (key, item) in item.into_iter() {
-                    data.push(key.into(), item);
+        pub struct MockDataTable {
+            items: HashMap<Vec<u8>, VecDeque<Message<Item>>>,
+        }
+
+        impl MockDataTable {
+            pub fn new() -> Self {
+                MockDataTable {
+                    items: HashMap::new(),
                 }
-                data
             }
 
-            fn push<I: Serialize>(&mut self, key: Vec<u8>, item: RpcResult<I>) {
-                let item = item.map(|x| serde_json::to_string(&x).unwrap());
+            pub fn push<I: Serialize>(&mut self, key: Vec<u8>, item: Message<RpcResult<I>>) {
+                let item = match item {
+                    Message::Many(items) => Message::Many(items.map(|items| {
+                        items
+                            .into_iter()
+                            .map(|item| item.map(|x| serde_json::to_string(&x).unwrap()))
+                            .collect()
+                    })),
+                    Message::Single(item) => {
+                        Message::Single(item.map(|x| serde_json::to_string(&x).unwrap()))
+                    }
+                };
                 match self.items.entry(key) {
```

Comment:
Nit: you can maybe simplify this a touch with something like:

```rust
self.items.entry(key).or_default().push_back(item)
```

or_default to put empty VecDeque in if empty, and then push the item to it in either case

Reply:
Done